### PR TITLE
chore: remove panic and continue with old config on incremental update failure

### PR DESCRIPTION
**Pull request by Sidddddarth**
# Description

incremental update failure causes a panic. Removing that panic.

## Linear Ticket

[Slack thread for context](https://rudderlabs.slack.com/archives/C01HTT66UMB/p1699502654415459)

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: backend-config/namespace_config.go [hunk 1]
```diff
@@ -139,10 +140,16 @@ func (nc *namespaceConfig) getFromAPI(ctx context.Context) (map[string]ConfigT,
 		if workspace == nil { // this workspace was not updated, populate it with the previous config
 			previousConfig, ok := nc.workspacesConfig[workspaceID]
 			if !ok {
-				panic(fmt.Errorf(
-					"workspace %q was not updated but was not present in previous config: %+v",
-					workspaceID, req,
-				))
+				nc.logger.Errorw(
+					"workspace was not updated but was not present in previous config",
```
Comment (atzoum) [inline backend-config/namespace_config.go:144]: we can just add an alert on this log right?

File: backend-config/backend-config.go [hunk 2]
```diff
@@ -177,6 +178,9 @@ func (bc *backendConfigImpl) configUpdate(ctx context.Context) {
 	if err != nil {
 		statConfigBackendError.Increment()
 		pkgLogger.Warnf("Error fetching config from backend: %v", err)
+		if errors.Is(err, ErrIncrementalUpdateFailed) {
+			stats.Default.NewStat("config_backend.ErrIncrementalUpdateFailed", stats.CountType).Increment()
+		}
```
Comment (atzoum) [inline backend-config/backend-config.go:183]: do we need another stat?

File: backend-config/namespace_config.go [hunk 3]
```diff
@@ -139,10 +140,16 @@ func (nc *namespaceConfig) getFromAPI(ctx context.Context) (map[string]ConfigT,
...
+					"workspaceID", workspaceID,
+					"req", req,
+				)
+				// reset state here
+				nc.lastUpdatedAt = time.Time{}
```
Comment (atzoum) [inline backend-config/namespace_config.go:149]: `namespaceConfig#Get` can react on the `ErrIncrementalUpdateFailed `by resetting the `lastUpdatedAt` and calling `getFromAPI` again. This way we won't have to wait for the next poll to happen

**Comment by atzoum** [comment]
If we want this to be released as a patch release ASAP it needs to target `1.16.x`, otherwise it will be included in the upcoming `1.17.x` release

**Comment by Sidddddarth** [comment]
> If we want this to be released as a patch release ASAP it needs to target `1.16.x`, otherwise it will be included in the upcoming `1.17.x` release


My bad
Opened [here](https://github.com/rudderlabs/rudder-server/pull/4109)

**Pull request closed without merging**